slotfills/sections/date-field.tsx

```diff
@@ -0,0 +1,126 @@
+  TextControl,
+  Button,
+} from '@wordpress/components';
+import { gmdateI18n, getDate, __experimentalGetSettings } from '@wordpress/date';
+import { __ } from '@wordpress/i18n';
+import Field from '@/interfaces/field';
+import FMObject from '@/interfaces/fm-object';
+
+import './date-field.scss';
+
+interface DateFieldProps {
+  field: Field;
+  index?: number;
+  label?: string;
+  valueHook: (key: number | string) => [string | FMObject | string[] | FMObject[], Function];
+}
+
+interface DateSettings {
+  timezone: {
+    offset: number;
+  }
+}
+
+export default function DateField({
+  field: {
+    attributes = {},
+    date_format: dateFormat = 'Y-m-d',
+    description = '',
+    description_after_element: descriptionAfterElement = true,
+    name,
+    use_time: useTime = false,
+  },
+  valueHook,
+  index = null,
+  label = '',
+}: DateFieldProps) {
+  const [showPicker, setShowPicker] = useState(false);
+  const [value, setValue] = index !== null ? valueHook(index) : valueHook(name);
+  let initialvalue = value && typeof value === 'object' && !Array.isArray(value) ? value[name] : value;
+  initialvalue = initialvalue ? String(initialvalue) : '';
+
+  const datesettings = __experimentalGetSettings() as DateSettings;
+  const {
+    timezone: {
+      offset = 0,
+    } = {},
+  } = datesettings;
+
+  const onChange = (newValue:string) => {
+    setValue((getDate(newValue).getTime() / 1000) + (offset * 60 * 60));
+    setShowPicker(false);
+  };
+
+  const onFieldChange = (newValue:string) => {
+    setValue((getDate(newValue).getTime() / 1000));
+    setShowPicker(false);
+  };
+
+  const dateFormatWithTime = useTime ? `${dateFormat} g:i a` : dateFormat;
+  const formattedDate = gmdateI18n(
+    dateFormatWithTime,
+    new Date(parseInt(initialvalue, 10) * 1000),
+  );
+
+  return (
+    <PanelRow>
+      <div className="fm-gutenberg-flex__column fm-gutenberg-popup-wrap">
+        {description && !descriptionAfterElement ? (
+          <div className="fm-gutenberg-item__description">{description}</div>
+        ) : null}
+        <label htmlFor={`${name}_${index}`}>
+          {label}
+        </label>
+        <TextControl
+          id={`${name}_${index}`}
+          value={formattedDate}
+          onClick={() => setShowPicker(!showPicker)}
+          onChange={onFieldChange}
+        />
+        {showPicker && !useTime ? (
+          <div className="fm-gutenberg-popup">
+            <Button
+              aria-label={__('Close', 'fm-gutenberg')}
+              onClick={() => setShowPicker(false)}
+              className="fm-gutenberg-popup__close"
+            >
+              x
+            </Button>
+            <DatePicker
+              {...attributes} // eslint-disable-line react/jsx-props-no-spreading
+              onChange={onChange}
+              currentDate={formattedDate}
+              key={`text-control-${name}-${index}`}
+            />
+          </div>
+        ) : null}
+        {showPicker && useTime ? (
```

Comment:
The only difference I see here is the `is12Hour`. Could you combine these and do a `is12Hour={useTime}`?

Reply:
@dustinpaluch its two different components: `<DatePicker>` and `<DateTimePicker>`

Reply:
Ah, got it!